pkg/watermark/publish/publisher.go

```diff
@@ -115,8 +115,9 @@ func (p *publish) PublishWatermark(wm processor.Watermark, offset isb.Offset) {
 
 	// build value
 	var seq int64
-	if p.opts.isSource || p.opts.isSink {
-		// For source and sink publisher, we don't care about the offset, also the sequence of the offset might not be integer.
+	if p.opts.isSource || p.opts.isSink || offset == nil {
```

Comment:
Do you see any case that offset is nil, but not a source or sink?

Reply:
No, I don't. I am adding this extra check to avoid NullPointerException below when we retrieve seq. But it could never be reached.

Reply:
Better not to add it, otherwise it will cover the problem when it really happens.

Reply:
That's a good point. Would it be better to add the check to beginning of this method? Something like below?

```
if !p.opts.isSource && !p.opts.isSink && offset == nil {
    p.log.Errorw("Unable to publish watermark. offset is nil but the publisher is neither source nor sink.")
}
```